compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -548,6 +548,10 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
                 }
             }
 
+            Rvalue::RawPtr(RawPtrKind::FakeForPtrMetadata, place) => {
+                assert!(!place.is_indirect(), "fake borrows are always indirect");
```

Comment:
```suggestion
                // These are only inserted for slice length, so the place must already be indirect.
                // This implies we do not have to worry about whether the borrow escapes.
                assert!(!place.is_indirect(), "fake borrows are always indirect");
```

Reply:
whoops the polarity of this assert is backwards too lol